src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/movements/MovementsService.kt

```diff
@@ -46,6 +48,27 @@ class MovementsService(
     )
   }
 
+  private fun List<OffenderBooking>.fixMergedSchedules() {
+    // find any scheduled returns on the wrong application and remove them (these are created during merges)
+    val scheduledReturnsWithWrongParent = this.flatMap { booking ->
+      booking.temporaryAbsenceApplications.map { application ->
+        application.scheduledTemporaryAbsence?.scheduledTemporaryAbsenceReturns
+          ?.filter { it.temporaryAbsenceApplication != null && it.temporaryAbsenceApplication != application }
+          ?.also { application.scheduledTemporaryAbsence?.scheduledTemporaryAbsenceReturns?.removeAll(it) }
+          ?: emptyList()
+      }
+    }.flatten()
+
+    // put the scheduled returns onto the correct application
+    scheduledReturnsWithWrongParent.forEach { mergedReturn ->
+      this.flatMap { booking -> booking.temporaryAbsenceApplications }
+        .find { it.movementApplicationId == mergedReturn.temporaryAbsenceApplication?.movementApplicationId }
+        ?.scheduledTemporaryAbsence
+        ?.scheduledTemporaryAbsenceReturns += mergedReturn
+      mergedReturn.temporaryAbsenceApplication = null
+    }
+  }
```

Comment:
Before mapping the schedules to the DTO, we move all of the corrupted scheduled returns onto the correct TAP applications, reinstating the one to one mapping.